neuro_san/internals/graph/activations/intra_agent_message_utils.py

```diff
@@ -11,14 +11,31 @@
 # END COPYRIGHT
 
 from typing import Any
+from typing import Dict
 from typing import List
+from typing import Type
 from typing import Union
 
 import json
 
+from langchain_core.messages.ai import AIMessage
 from langchain_core.messages.base import BaseMessage
+from langchain_core.messages.human import HumanMessage
+from langchain_core.messages.system import SystemMessage
 
-from neuro_san.internals.messages.chat_message_type import ChatMessageType
+from neuro_san.internals.messages.agent_framework_message import AgentFrameworkMessage
+from neuro_san.internals.messages.agent_message import AgentMessage
+from neuro_san.internals.messages.agent_tool_result_message import AgentToolResultMessage
+
+
+_MESSAGE_TYPE_TO_ROLE: Dict[Type[BaseMessage], str] = {
+    AIMessage: "assistant",
+    HumanMessage: "user",
+    SystemMessage: "system",
+    AgentMessage: "agent",
+    AgentFrameworkMessage: "agent-framework",
+    AgentToolResultMessage: "agent-tool-result",
+}
```

Comment:
Move this message -> role lookup from ChatMessageType into IntraAgentMessageUtils, which is the only place that uses it.  This only is used by _message_to_role() now moved to this class below.